provers/sp1/driver/src/lib.rs

```diff
@@ -70,7 +135,17 @@ impl Prover for Sp1Prover {
         .into())
     }
 
-    async fn cancel(_key: ProofKey, _store: Box<&mut dyn IdStore>) -> ProverResult<()> {
+    async fn cancel(key: ProofKey, store: Box<&mut dyn IdStore>) -> ProverResult<()> {
+        let proof_id = store.read_id(key)?;
```

Comment:
to be clear, current cancel is for remote task, not for local. and eventually we will have both remote & local, is it?

Reply:
Yes, this is just mocked up to avoid to many conflicts with the distributed prover that is coming to sp1